### PR TITLE
17427 - Upgrade to Typescript 4.9.5 + Linting Rules

### PR DESCRIPTION
**Pull request by seeker25**
https://github.com/bcgov/entity/issues/17427

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

**Comment by seeker25** [comment]
I'll take a look at the code smells, security hotspots, bugs tomorrow - many of these are preexisting and some are false positives.

File: auth-web/src/components/auth/account-settings/team-management/AddUsersForm.vue [hunk 1]
```diff
@@ -92,12 +114,11 @@
 <script lang="ts">
 import { AddUserBody, AddUsersToOrgBody, Member, MembershipType, Organization, RoleInfo } from '@/models/Organization'
 import { Component, Emit, Vue } from 'vue-property-decorator'
-import { mapActions, mapMutations, mapState } from 'vuex'
+import { mapActions, mapState } from 'vuex'
 import CommonUtils from '@/util/common-util'
-import { Invitation } from '@/models/Invitation'
+/* eslint-disable-next-line @typescript-eslint/no-unused-vars */
```
Comment (seeker25) [inline auth-web/src/components/auth/account-settings/team-management/AddUsersForm.vue:119]: Will be removing this with Pinia

File: auth-web/src/components/auth/account-settings/statement/StatementsSettings.vue [hunk 2]
```diff
@@ -160,31 +178,33 @@
...
 </template>
 
 <script lang="ts">
-import { Component, Mixins, Prop, Vue, Watch } from 'vue-property-decorator'
+import { Component, Vue } from 'vue-property-decorator'
 import { Member, MembershipType, Organization } from '@/models/Organization'
 import { StatementListItem, StatementNotificationSettings, StatementRecipient, StatementSettings } from '@/models/statement'
 import { mapActions, mapState } from 'vuex'
 import CommonUtils from '@/util/common-util'
+/* eslint-disable-next-line @typescript-eslint/no-unused-vars */
```
Comment (seeker25) [inline auth-web/src/components/auth/account-settings/statement/StatementsSettings.vue:207]: If moment isn't included for some of these components I remember seeing runtime errors.

File: auth-web/src/components/auth/BusinessContactForm.vue [hunk 3]
```diff
@@ -1,100 +1,120 @@
...
+      </v-btn>
+    </div>
   </v-form>
 </template>
 
 <script lang="ts">
 import { Business, FolioNumberload } from '@/models/business'
 import { Component, Vue } from 'vue-property-decorator'
 import { mapActions, mapState } from 'vuex'
+/* eslint-disable-next-line @typescript-eslint/no-unused-vars */
```
Comment (seeker25) [inline auth-web/src/components/auth/BusinessContactForm.vue:113]: Will remove with Pinia

**Review by hfekete — APPROVED** [review]
Nice!

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-account-dev--pr-2459-0teophvy.web.app

**Comment by Jxio** [comment]
image
can't view above temporary url

**Comment by seeker25** [comment]
weird, runs locally

File: auth-web/src/components/auth/account-settings/account-info/AccountInfo.vue [hunk 4]
```diff
@@ -157,7 +157,7 @@
             <!-- TODO: can use v-can instead of v-if if all user with change permisson have view also -->
             <AccountMailingAddress
               ref="mailingAddress"
-              :baseAddress="baseAddress"
+              :baseAddress="originalAddress"
```
[the remark this answers is not in the document]
Comment (seeker25) [reply auth-web/src/components/auth/account-settings/account-info/AccountInfo.vue:160]: I tried a couple - I'll look over them again

[the remark this answers is not in the document]
Comment (JazzarKarim) [reply auth-web/src/components/auth/account-settings/account-info/AccountInfo.vue:160]: Awesome! Resolving

[the remark this answers is not in the document]
Comment (seeker25) [reply auth-web/src/components/auth/account-settings/account-info/AccountInfo.vue:160]: image

**Comment by JazzarKarim** [comment]
Is it running locally for you Travis?
config null

**Comment by seeker25** [comment]
Not running under prod build:

`npm run build && npm run preview`

Seems weird, looking into it still

**Review by Jxio — APPROVED** [review]

**Comment by seeker25** [comment]
Should be fixed now, we'll see

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-account-dev--pr-2459-0teophvy.web.app

**Comment by seeker25** [comment]
Success!

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-account-dev--pr-2459-0teophvy.web.app

**Comment by JazzarKarim** [comment]
> Success!

Awesome!!! I'm looking at the commit, I see you had to change the i18n package 🙌

**Comment by seeker25** [comment]
Yup, that package was a huge pain in the neck. It's an international library, but we only use english anyways.. but it caused alot of pain.

**Comment by JazzarKarim** [comment]
> Yup, that package was a huge pain in the neck. It's an international library, but we only use english anyways.. but it caused alot of pain.

Yup, very understandable.

**Review by JazzarKarim — APPROVED** [review]
Looks great Travis! 🥳

**Pull request merged**